sktime/transformations/series/feature_selection.py

```diff
@@ -0,0 +1,196 @@
+
+        # multivariate Z
+        if not isinstance(Z, pd.Series):
+            if self.method == "feature-importances":
+                self.regressor_ = check_regressor(
+                    regressor=self.regressor, random_state=self.random_state
+                )
+                self._check_n_columns(Z)
+                X = check_series(X)
+                # fit regressor with Z as exog data and X as endog data (target)
+                self.regressor_.fit(X=Z, y=X)
+                if not hasattr(self.regressor_, "feature_importances_"):
+                    raise ValueError(
+                        """The given regressor must have an
+                        attribute feature_importances_ after fitting."""
+                    )
+                # create dict with columns name (key) and feauter importance (value)
+                d = dict(zip(Z.columns, self.regressor_.feature_importances_))
+                # sort d descending
+                d = {k: d[k] for k in sorted(d, key=d.get, reverse=True)}
+                self.feature_importances_ = d
+                self.columns_ = list(d.keys())[: self.n_columns_]
+            elif self.method == "random":
+                self._check_n_columns(Z)
+                self.columns_ = list(
+                    Z.sample(
+                        n=self.n_columns_, random_state=self.random_state, axis=1
+                    ).columns
+                )
+            elif self.method == "columns":
+                if self.columns is None:
+                    raise AttributeError("Parameter columns must be given.")
+                self.columns_ = self.columns
+            elif self.method == "none":
+                self.columns_ = None
+            elif self.method == "all":
+                self.columns_ = list(Z.columns)
+            else:
+                raise ValueError("Incorrect method given. Try another method.")
+
+        self._is_fitted = True
+        return self
+
+    def transform(self, Z, X=None):
+        """Return transformed version of input series `Z`.
+
+        Parameters
+        ----------
+        Z : pd.DataFrame
+            A time series to apply the transformation on.
+
+        Returns
+        -------
+        Zt : pd.Series or pd.DataFrame
+            Transformed version of input series `Z`.
+        """
+        self.check_is_fitted()
+        Z = check_series(Z)
+
+        # multivariate case
+        if not isinstance(Z, pd.Series):
+            if self.method == "none":
+                Zt = None
+            else:
+                Zt = Z[self.columns_]
+        # univariate case
+        else:
+            if self.method == "none":
+                Zt = None
+            else:
+                Zt = Z
+        return Zt
+
+    def _check_n_columns(self, Z):
+        if not isinstance(self.n_columns_, int):
+            self.n_columns_ = int(math.ceil(Z.shape[1] / 2))
+
+    @classmethod
+    def get_test_params(cls):
```

Comment:
Shouldn't this be implemented in the base class? The test parameters should then be a class variable similar to the tags.

Reply:
I couldnt find how to actually implement this. It says its in the extension template but there I cant find it.

Reply:
that's because we separated the refactor (moving logic) from adding the documentation in the extension templates.

The extension template PR is still a draft and under review here:
https://github.com/alan-turing-institute/sktime/pull/1395

The docstring should say though what it should do:
return either a dict of params or a list of dict; in either case, the dicts must be suitable for passing to the init, i.e., `__init__(**mydict)` must result in a sensible test instance for the class.

Reply:
FYI @Aparna-Sakshi

Reply:
(would be nice to finish the extension template PR 😃 )

Reply:
besides, you can still use `_config` as before if it would be a blocker for this PR

Reply:
yes will just use `_config` to get this PR done :)

Reply:
Well, if your line in `_config` looks like this:
`FeatureSelectionTransformer: {"foo": "bar"}`

then the content of `get_test_params` should look like this:
`return {"foo": "bar"}`

Should be straightforward...